Microsoft Windows [Version 10.0.19041.630]
(c) 2020 Microsoft Corporation. All rights reserved.

C:\Users\New>py
Python 3.8.3 (tags/v3.8.3:6f8c832, May 13 2020, 22:37:02) [MSC v.1924 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> from opcua import Client
>>> client = Client("opc.tcp://127.0.0.1:49330")
>>> client.connect()
Requested session timeout to be 3600000ms, got 60000ms instead
>>> client.connect()
>>> client.get_children()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'Client' object has no attribute 'get_children'
>>> objects = client.get_objects_node()
>>> objects.get_children()
[Node(FourByteNodeId(i=2253)), Node(StringNodeId(ns=2;s=_AdvancedTags)), Node(StringNodeId(ns=2;s=_ConnectionSharing)), Node(StringNodeId(ns=2;s=_CustomAlarms)), Node(StringNodeId(ns=2;s=_DataLogger)), Node(StringNodeId(ns=2;s=_EFMExporter)), Node(StringNodeId(ns=2;s=_IDF_for_Splunk)), Node(StringNodeId(ns=2;s=_IoT_Gateway)), Node(StringNodeId(ns=2;s=_LocalHistorian)), Node(StringNodeId(ns=2;s=_Redundancy)), Node(StringNodeId(ns=2;s=_Scheduler)), Node(StringNodeId(ns=2;s=_SecurityPolicies)), Node(StringNodeId(ns=2;s=_SNMP Agent)), Node(StringNodeId(ns=2;s=_System)), Node(StringNodeId(ns=2;s=_ThingWorx)), Node(StringNodeId(ns=2;s=event_chek)), Node(StringNodeId(ns=2;s=simu))]
>>> simu = client.get_children()[-1]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'Client' object has no attribute 'get_children'
>>> simu = objects.get_children()[-1]
>>> simu
Node(StringNodeId(ns=2;s=simu))
>>> while True:
... simu.get_value()
  File "<stdin>", line 2
    simu.get_value()
    ^
IndentationError: expected an indented block
>>> while True:
...     simu.get_value()
...     print(simu.get_value())
...
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
  File "C:\Users\New\AppData\Local\Programs\Python\Python38\lib\site-packages\opcua\common\node.py", line 155, in get_value
    result = self.get_data_value()
  File "C:\Users\New\AppData\Local\Programs\Python\Python38\lib\site-packages\opcua\common\node.py", line 164, in get_data_value
    return self.get_attribute(ua.AttributeIds.Value)
  File "C:\Users\New\AppData\Local\Programs\Python\Python38\lib\site-packages\opcua\common\node.py", line 276, in get_attribute
    result[0].StatusCode.check()
  File "C:\Users\New\AppData\Local\Programs\Python\Python38\lib\site-packages\opcua\ua\uatypes.py", line 218, in check
    raise UaStatusCodeError(self.value)
opcua.ua.uaerrors._auto.BadAttributeIdInvalid: "The attribute is not supported for the specified Node."(BadAttributeIdInvalid)
>>> simu.get_value
<bound method Node.get_value of Node(StringNodeId(ns=2;s=simu))>
>>> simu.get_value()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\New\AppData\Local\Programs\Python\Python38\lib\site-packages\opcua\common\node.py", line 155, in get_value
    result = self.get_data_value()
  File "C:\Users\New\AppData\Local\Programs\Python\Python38\lib\site-packages\opcua\common\node.py", line 164, in get_data_value
    return self.get_attribute(ua.AttributeIds.Value)
  File "C:\Users\New\AppData\Local\Programs\Python\Python38\lib\site-packages\opcua\common\node.py", line 276, in get_attribute
    result[0].StatusCode.check()
  File "C:\Users\New\AppData\Local\Programs\Python\Python38\lib\site-packages\opcua\ua\uatypes.py", line 218, in check
    raise UaStatusCodeError(self.value)
opcua.ua.uaerrors._auto.BadAttributeIdInvalid: "The attribute is not supported for the specified Node."(BadAttributeIdInvalid)
>>> simu_sens = client.get_node("ns=2;s=simu)
  File "<stdin>", line 1
    simu_sens = client.get_node("ns=2;s=simu)
                                            ^
SyntaxError: EOL while scanning string literal
>>>
KeyboardInterrupt
>>> simu_sens = client.get_node("ns=2;s=simu")
>>> simu_sens.get_value()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\New\AppData\Local\Programs\Python\Python38\lib\site-packages\opcua\common\node.py", line 155, in get_value
    result = self.get_data_value()
  File "C:\Users\New\AppData\Local\Programs\Python\Python38\lib\site-packages\opcua\common\node.py", line 164, in get_data_value
    return self.get_attribute(ua.AttributeIds.Value)
  File "C:\Users\New\AppData\Local\Programs\Python\Python38\lib\site-packages\opcua\common\node.py", line 276, in get_attribute
    result[0].StatusCode.check()
  File "C:\Users\New\AppData\Local\Programs\Python\Python38\lib\site-packages\opcua\ua\uatypes.py", line 218, in check
    raise UaStatusCodeError(self.value)
opcua.ua.uaerrors._auto.BadAttributeIdInvalid: "The attribute is not supported for the specified Node."(BadAttributeIdInvalid)
>>> simu_sens
Node(StringNodeId(ns=2;s=simu))
>>> simu_sens = client.get_node("ns=2;s=simu.Device1.tG")
>>> simu_sens
Node(StringNodeId(ns=2;s=simu.Device1.tG))
>>> simu_sens.get_value()
9720
>>> while True:
...     print(simu_sens.get_value())
...
9845
9846
9847
9848
9849
9850
9851
9852
9853
9854
9856
9857
9858
9859
9860
9861
9862
9863
9864
9865
9866
9867
9868
9869
9870
9871
9873
9874
9875
9876
9877
9878
9880
9881
9882
9883
9884
9885
9886
9887
9888
9889
9891
9892
9893
9894
9895
9896
9897
9898
9899
9900
9901
9902
9903
9904
9905
9906
9908
9909
9910
9911
9912
9913
9914
9915
9916
9917
9918
9919
9920
9921
9922
9924
9925
9926
9927
9928
9929
9930
9931
9932
9933
9934
9935
9936
9937
9938
9939
9941
9942
9943
9944
9945
9946
9948
9949
9950
9951
9952
9953
9954
9955
9956
9957
9959
9960
9961
9962
9963
9964
9965
9966
9967
9968
9969
9970
9971
9972
9973
9974
9976
9977
9978
9979
9980
9981
9982
9983
9984
9985
9986
9987
9988
9989
9990
9991
9993
9994
9995
9996
9997
9998
9999
10000
10001
10002
10003
10004
10005
10006
10007
10008
10010
10011
10012
10013
10014
10015
10017
10018
10019
10020
10021
10022
10023
10024
10025
10026
10028
10029
10030
10031
10032
10033
10034
10035
10036
10037
10038
10039
10040
10041
10042
10043
10045
10046
10047
10048
10049
10050
10051
10052
10053
10054
10055
10056
10057
10058
10059
10060
10062
10063
10064
10065
10066
10067
10068
10069
10070
10071
10072
10073
10074
10075
10076
10078
10079
10080
10081
10082
10083
10085
10086
10087
10088
10089
10090
10091
10092
10093
10094
10096
10097
10098
10099
10100
10101
10102
10103
10104
10105
10106
10107
10108
10109
10110
10111
10113
10114
10115
10116
10117
10118
10119
10120
10121
10122
10123
10124
10125
10126
10127
10128
10130
10131
10132
10133
10134
10135
10136
10137
10138
10139
10140
10141
10142
10143
10144
10145
10147
10148
10149
10150
10151
10152
10154
10155
10156
10157
10158
10159
10160
10161
10162
10163
10165
10166
10167
10168
10169
10170
10171
10172
10173
10174
10175
10176
10177
10178
10179
10180
10182
10183
10184
10185
10186
10187
10188
10189
10190
10191
10192
10193
10194
10195
10196
10197
10199
10200
10201
10202
10203
10204
10205
10206
10207
10208
10209
10210
10211
10212
10213
10214
10216
10217
10218
10219
10220
10222
10223
10224
10225
10226
10227
10228
10229
10230
10231
10233
10234
10235
10236
10237
10238
10239
10240
10241
10242
10243
10244
10245
10246
10247
10248
10250
10251
10252
10253
10254
10255
10256
10257
10258
10259
10260
10261
10262
10263
10264
10265
10267
10268
10269
10270
10271
10272
10273
10274
10275
10276
10277
10278
10279
10280
10281
10282
10284
10285
10286
10287
10288
10289
10291
10292
10293
10294
10295
10296
10297
10298
10299
10300
10302
10303
10304
10305
10306
10307
10308
10309
10310
10311
10312
10313
10314
10315
10316
10317
10319
10320
10321
10322
10323
10324
10325
10326
10327
10328
10329
10330
10331
10332
10334
10335
10336
10337
10338
10339
10340
10341
10342
10343
10344
10345
10346
10347
10348
10349
10351
10352
10353
10354
10355
10356
10358
10359
10360
10361
10362
10363
10364
10365
10366
10367
10369
10370
10371
10372
10373
10374
10375
10376
10377
10378
10379
10380
10381
10382
10383
10384
10386
10387
10388
10389
10390
10391
10392
10393
10394
10395
10396
10397
10398
10399
10400
10402
10403
10404
10405
10406
10407
10408
10409
10410
10411
10412
10413
10414
10415
10416
10418
10419
10420
10421
10422
10423
10425
10426
10427
10428
10429
10430
10431
10432
10433
10434
10435
10437
10438
10439
10440
10441
10442
10443
10444
10445
10446
10447
10448
10449
10450
10451
10453
10454
10455
10456
10457
10458
10459
10460
10461
10462
10463
10464
10465
10466
10468
10469
10470
10471
10472
10473
10474
10475
10476
10477
10478
10479
10480
10481
10482
10483
10485
10486
10487
10488
10489
10490
10492
10493
10494
10495
10496
10497
10498
10499
10500
10501
10503
10504
10505
10506
10507
10508
10509
10510
10511
10512
10513
10514
10515
10516
10517
10518
10520
10521
10522
10523
10524
10525
10526
10527
10528
10529
10530
10531
10532
10533
10534
10535
10537
10538
10539
10540
10541
10542
10543
10544
10545
10546
10547
10548
10549
10550
10551
10552
10554
10555
10556
10557
10558
10559
10561
10562
10563
10564
10565
10566
10567
10568
10569
10570
10572
10573
10574
10575
10576
10577
10578
10579
10580
10581
10582
10583
10584
10585
10586
10587
10589
10590
10591
10592
10593
10594
10595
10596
10597
10598
10599
10600
10601
10602
10603
10604
10606
10607
10608
10609
10610
10611
10612
10613
10614
10615
10616
10617
10618
10619
10620
10622
10623
10624
10625
10626
10627
10629
10630
10631
10632
10633
10634
10635
10636
10637
10638
10640
10641
10642
10643
10644
10645
10646
10647
10648
10649
10650
10651
10652
10653
10654
10655
10657
10658
10659
10660
10661
10662
10663
10664
10665
10666
10667
10668
10669
10670
10671
10672
10674
10675
10676
10677
10678
10679
10680
10681
10682
10683
10684
10685
10686
10687
10688
10689
10691
10692
10693
10694
10695
10696
10698
10699
10700
10701
10702
10703
10704
10705
10706
10708
10709
10710
10711
10712
10713
10714
10715
10716
10717
10718
10719
10720
10721
10722
10723
10725
10726
10727
10728
10729
10730
10731
10732
10733
10734
10735
10736
10737
10738
10739
10740
10742
10743
10744
10745
10746
10747
10748
10749
10750
10751
10752
10753
10754
10755
10756
10757
10759
10760
10761
10762
10763
10764
10766
10767
10768
10769
10770
10771
10772
10773
10774
10775
10777
10778
10779
10780
10781
10782
10783
10784
10785
10786
10787
10788
10789
10790
10791
10793
10794
10795
10796
10797
10798
10799
10800
10801
10802
10803
10804
10805
10806
10807
10808
10810
10811
10812
10813
10814
10815
10816
10817
10818
10819
10820
10821
10822
10823
10824
10825
10827
10828
10829
10830
10831
10832
10834
10835
10836
10837
10838
10839
10840
10841
10842
10843
10845
10846
10847
10848
10849
10850
10851
10852
10853
10854
10855
10856
10857
10858
10859
10860
10862
10863
10864
10865
10866
10867
10868
10869
10870
10871
10872
10873
10874
10875
10876
10877
10879
10880
10881
10882
10883
10884
10885
10886
10887
10888
10889
10890
10891
10892
10893
10894
10896
10897
10898
10899
10900
10901
10903
10904
10905
10906
10907
10908
10909
10910
10911
10912
10914
10915
10916
10917
10918
10919
10920
10921
10922
10923
10924
10925
10926
10927
10928
10929
10931
10932
10933
10934
10943
10944
10945
10946
10947
10949
10950
10951
10952
10953
10954
10955
10956
10957
10958
10959
10960
10961
10962
10963
10965
10966
10967
10968
10969
10970
10971
10972
10973
10974
10975
10976
10977
10978
10979
10980
10982
10983
10984
10985
10986
10987
10988
10989
10990
10991
10992
10993
10994
10995
10996
10997
10999
11000
11001
11002
11003
11004
11006
11007
11008
11009
11010
11011
11012
11013
11014
11015
11017
11018
11019
11020
11021
11022
11023
11024
11025
11026
11027
11028
11029
11030
11031
11032
11034
11035
11036
11037
11038
11039
11040
11041
11042
11043
11044
11045
11046
11047
11048
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
  File "C:\Users\New\AppData\Local\Programs\Python\Python38\lib\site-packages\opcua\common\node.py", line 155, in get_value
    result = self.get_data_value()
  File "C:\Users\New\AppData\Local\Programs\Python\Python38\lib\site-packages\opcua\common\node.py", line 164, in get_data_value
    return self.get_attribute(ua.AttributeIds.Value)
  File "C:\Users\New\AppData\Local\Programs\Python\Python38\lib\site-packages\opcua\common\node.py", line 275, in get_attribute
    result = self.server.read(params)
  File "C:\Users\New\AppData\Local\Programs\Python\Python38\lib\site-packages\opcua\client\ua_client.py", line 347, in read
    data = self._uasocket.send_request(request)
  File "C:\Users\New\AppData\Local\Programs\Python\Python38\lib\site-packages\opcua\client\ua_client.py", line 83, in send_request
    data = future.result(self.timeout)
  File "C:\Users\New\AppData\Local\Programs\Python\Python38\lib\concurrent\futures\_base.py", line 434, in result
    self._condition.wait(timeout)
  File "C:\Users\New\AppData\Local\Programs\Python\Python38\lib\threading.py", line 306, in wait
    gotit = waiter.acquire(True, timeout)
KeyboardInterrupt
>>> simu_sens.get_value()
23006
>>> objects.get_children()
[Node(FourByteNodeId(i=2253)), Node(StringNodeId(ns=2;s=_AdvancedTags)), Node(StringNodeId(ns=2;s=_ConnectionSharing)), Node(StringNodeId(ns=2;s=_CustomAlarms)), Node(StringNodeId(ns=2;s=_DataLogger)), Node(StringNodeId(ns=2;s=_EFMExporter)), Node(StringNodeId(ns=2;s=_IDF_for_Splunk)), Node(StringNodeId(ns=2;s=_IoT_Gateway)), Node(StringNodeId(ns=2;s=_LocalHistorian)), Node(StringNodeId(ns=2;s=_Redundancy)), Node(StringNodeId(ns=2;s=_Scheduler)), Node(StringNodeId(ns=2;s=_SecurityPolicies)), Node(StringNodeId(ns=2;s=_SNMP Agent)), Node(StringNodeId(ns=2;s=_System)), Node(StringNodeId(ns=2;s=_ThingWorx)), Node(StringNodeId(ns=2;s=event_chek)), Node(StringNodeId(ns=2;s=simu))]
>>> objects.get_children()=as
  File "<stdin>", line 1
    objects.get_children()=as
                           ^
SyntaxError: invalid syntax
>>> as = objects.get_children()
  File "<stdin>", line 1
    as = objects.get_children()
    ^
SyntaxError: invalid syntax
>>> as1 = objects.get_children()
>>> as1
[Node(FourByteNodeId(i=2253)), Node(StringNodeId(ns=2;s=_AdvancedTags)), Node(StringNodeId(ns=2;s=_ConnectionSharing)), Node(StringNodeId(ns=2;s=_CustomAlarms)), Node(StringNodeId(ns=2;s=_DataLogger)), Node(StringNodeId(ns=2;s=_EFMExporter)), Node(StringNodeId(ns=2;s=_IDF_for_Splunk)), Node(StringNodeId(ns=2;s=_IoT_Gateway)), Node(StringNodeId(ns=2;s=_LocalHistorian)), Node(StringNodeId(ns=2;s=_Redundancy)), Node(StringNodeId(ns=2;s=_Scheduler)), Node(StringNodeId(ns=2;s=_SecurityPolicies)), Node(StringNodeId(ns=2;s=_SNMP Agent)), Node(StringNodeId(ns=2;s=_System)), Node(StringNodeId(ns=2;s=_ThingWorx)), Node(StringNodeId(ns=2;s=event_chek)), Node(StringNodeId(ns=2;s=simu))]
>>> as1.get_children()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'list' object has no attribute 'get_children'
>>> as1[-1].get_children()
[Node(StringNodeId(ns=2;s=simu._Statistics)), Node(StringNodeId(ns=2;s=simu._System)), Node(StringNodeId(ns=2;s=simu.Device1))]
>>> as1[-1].get_children()[-1].get_children()
[Node(StringNodeId(ns=2;s=simu.Device1._System)), Node(StringNodeId(ns=2;s=simu.Device1.tG))]
>>> as1[-1].get_children()[-1].get_children()[-1].get_value()
23634
>>> as1[-1].get_children()[-1].get_children()[0].get_value()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\New\AppData\Local\Programs\Python\Python38\lib\site-packages\opcua\common\node.py", line 155, in get_value
    result = self.get_data_value()
  File "C:\Users\New\AppData\Local\Programs\Python\Python38\lib\site-packages\opcua\common\node.py", line 164, in get_data_value
    return self.get_attribute(ua.AttributeIds.Value)
  File "C:\Users\New\AppData\Local\Programs\Python\Python38\lib\site-packages\opcua\common\node.py", line 276, in get_attribute
    result[0].StatusCode.check()
  File "C:\Users\New\AppData\Local\Programs\Python\Python38\lib\site-packages\opcua\ua\uatypes.py", line 218, in check
    raise UaStatusCodeError(self.value)
opcua.ua.uaerrors._auto.BadAttributeIdInvalid: "The attribute is not supported for the specified Node."(BadAttributeIdInvalid)
>>> as1[-1].get_children()[-1].get_children()[0].get_value()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\New\AppData\Local\Programs\Python\Python38\lib\site-packages\opcua\common\node.py", line 155, in get_value
    result = self.get_data_value()
  File "C:\Users\New\AppData\Local\Programs\Python\Python38\lib\site-packages\opcua\common\node.py", line 164, in get_data_value
    return self.get_attribute(ua.AttributeIds.Value)
  File "C:\Users\New\AppData\Local\Programs\Python\Python38\lib\site-packages\opcua\common\node.py", line 276, in get_attribute
    result[0].StatusCode.check()
  File "C:\Users\New\AppData\Local\Programs\Python\Python38\lib\site-packages\opcua\ua\uatypes.py", line 218, in check
    raise UaStatusCodeError(self.value)
opcua.ua.uaerrors._auto.BadAttributeIdInvalid: "The attribute is not supported for the specified Node."(BadAttributeIdInvalid)
>>> objects = client.get_root_node()
>>> objects
Node(TwoByteNodeId(i=84))
>>> objects.get_children()
[Node(TwoByteNodeId(i=85)), Node(TwoByteNodeId(i=86)), Node(TwoByteNodeId(i=87))]
>>> objects = client.get_objects_node()
>>> objects
Node(TwoByteNodeId(i=85))
>>> objects.get_children()
[Node(FourByteNodeId(i=2253)), Node(StringNodeId(ns=2;s=_AdvancedTags)), Node(StringNodeId(ns=2;s=_ConnectionSharing)), Node(StringNodeId(ns=2;s=_CustomAlarms)), Node(StringNodeId(ns=2;s=_DataLogger)), Node(StringNodeId(ns=2;s=_EFMExporter)), Node(StringNodeId(ns=2;s=_IDF_for_Splunk)), Node(StringNodeId(ns=2;s=_IoT_Gateway)), Node(StringNodeId(ns=2;s=_LocalHistorian)), Node(StringNodeId(ns=2;s=_Redundancy)), Node(StringNodeId(ns=2;s=_Scheduler)), Node(StringNodeId(ns=2;s=_SecurityPolicies)), Node(StringNodeId(ns=2;s=_SNMP Agent)), Node(StringNodeId(ns=2;s=_System)), Node(StringNodeId(ns=2;s=_ThingWorx)), Node(StringNodeId(ns=2;s=event_chek)), Node(StringNodeId(ns=2;s=simu))]
>>> chil = objects.get_children()
>>> chil_simu = chil[-1].get_child()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: get_child() missing 1 required positional argument: 'path'
>>> chil_simu = chil[-1].get_children()
>>> chil_simu
[Node(StringNodeId(ns=2;s=simu._Statistics)), Node(StringNodeId(ns=2;s=simu._System)), Node(StringNodeId(ns=2;s=simu.Device1))]
>>> client = Client("opc.tcp://127.0.0.1:49330")
>>> objects = client.get_objects_node()
>>> chil = objects.get_children()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\New\AppData\Local\Programs\Python\Python38\lib\site-packages\opcua\common\node.py", line 316, in get_children
    return self.get_referenced_nodes(refs, ua.BrowseDirection.Forward, nodeclassmask)
  File "C:\Users\New\AppData\Local\Programs\Python\Python38\lib\site-packages\opcua\common\node.py", line 390, in get_referenced_nodes
    references = self.get_references(refs, direction, nodeclassmask, includesubtypes)
  File "C:\Users\New\AppData\Local\Programs\Python\Python38\lib\site-packages\opcua\common\node.py", line 369, in get_references
    results = self.server.browse(params)
  File "C:\Users\New\AppData\Local\Programs\Python\Python38\lib\site-packages\opcua\client\ua_client.py", line 327, in browse
    data = self._uasocket.send_request(request)
AttributeError: 'NoneType' object has no attribute 'send_request'
>>> objects = client.get_objects_node()
>>> objects
Node(TwoByteNodeId(i=85))
>>> objects.get_children()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\New\AppData\Local\Programs\Python\Python38\lib\site-packages\opcua\common\node.py", line 316, in get_children
    return self.get_referenced_nodes(refs, ua.BrowseDirection.Forward, nodeclassmask)
  File "C:\Users\New\AppData\Local\Programs\Python\Python38\lib\site-packages\opcua\common\node.py", line 390, in get_referenced_nodes
    references = self.get_references(refs, direction, nodeclassmask, includesubtypes)
  File "C:\Users\New\AppData\Local\Programs\Python\Python38\lib\site-packages\opcua\common\node.py", line 369, in get_references
    results = self.server.browse(params)
  File "C:\Users\New\AppData\Local\Programs\Python\Python38\lib\site-packages\opcua\client\ua_client.py", line 327, in browse
    data = self._uasocket.send_request(request)
AttributeError: 'NoneType' object has no attribute 'send_request'
>>> objects
Node(TwoByteNodeId(i=85))
>>> ob_chil = objects.get_children()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\New\AppData\Local\Programs\Python\Python38\lib\site-packages\opcua\common\node.py", line 316, in get_children
    return self.get_referenced_nodes(refs, ua.BrowseDirection.Forward, nodeclassmask)
  File "C:\Users\New\AppData\Local\Programs\Python\Python38\lib\site-packages\opcua\common\node.py", line 390, in get_referenced_nodes
    references = self.get_references(refs, direction, nodeclassmask, includesubtypes)
  File "C:\Users\New\AppData\Local\Programs\Python\Python38\lib\site-packages\opcua\common\node.py", line 369, in get_references
    results = self.server.browse(params)
  File "C:\Users\New\AppData\Local\Programs\Python\Python38\lib\site-packages\opcua\client\ua_client.py", line 327, in browse
    data = self._uasocket.send_request(request)
AttributeError: 'NoneType' object has no attribute 'send_request'
>>> client.connect()
Requested session timeout to be 3600000ms, got 60000ms instead
>>> client.connect()
>>> ob_chil = objects.get_children()
>>> ob_chil
[Node(FourByteNodeId(i=2253)), Node(StringNodeId(ns=2;s=_AdvancedTags)), Node(StringNodeId(ns=2;s=_ConnectionSharing)), Node(StringNodeId(ns=2;s=_CustomAlarms)), Node(StringNodeId(ns=2;s=_DataLogger)), Node(StringNodeId(ns=2;s=_EFMExporter)), Node(StringNodeId(ns=2;s=_IDF_for_Splunk)), Node(StringNodeId(ns=2;s=_IoT_Gateway)), Node(StringNodeId(ns=2;s=_LocalHistorian)), Node(StringNodeId(ns=2;s=_Redundancy)), Node(StringNodeId(ns=2;s=_Scheduler)), Node(StringNodeId(ns=2;s=_SecurityPolicies)), Node(StringNodeId(ns=2;s=_SNMP Agent)), Node(StringNodeId(ns=2;s=_System)), Node(StringNodeId(ns=2;s=_ThingWorx)), Node(StringNodeId(ns=2;s=event_chek)), Node(StringNodeId(ns=2;s=simu))]
>>> last_simu = ob_chil[-1].get_children()
>>> last_simu
[Node(StringNodeId(ns=2;s=simu._Statistics)), Node(StringNodeId(ns=2;s=simu._System)), Node(StringNodeId(ns=2;s=simu.Device1))]
>>> grp_data = last_simu[-1].get_children()
>>> grp_data
[Node(StringNodeId(ns=2;s=simu.Device1._System)), Node(StringNodeId(ns=2;s=simu.Device1.Group1)), Node(StringNodeId(ns=2;s=simu.Device1.tG))]
>>> grp_data_dev1 = grp_data[1].get_children()
>>> grp_data_dev1
[Node(StringNodeId(ns=2;s=simu.Device1.Group1.Meter)), Node(StringNodeId(ns=2;s=simu.Device1.Group1.Press)), Node(StringNodeId(ns=2;s=simu.Device1.Group1.temp))]
>>> len(grp_data_dev1)
3
>>> for i in range(len(grp_data_dev1)):
...     print(i.get_value())
...
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
AttributeError: 'int' object has no attribute 'get_value'
>>> for i in range(len(grp_data_dev1)):
... ...     print(for i in range(len(grp_data_dev1)):
  File "<stdin>", line 2
    ...     print(for i in range(len(grp_data_dev1)):
    ^
IndentationError: expected an indented block
>>> ...     print(i.get_value())
  File "<stdin>", line 1
    ...     print(i.get_value())
            ^
SyntaxError: invalid syntax
>>> for i in range(3):
...     print(grp_data_dev1.get_child(i))
...
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
AttributeError: 'list' object has no attribute 'get_child'
>>> for i in range(3):
...
  File "<stdin>", line 2

    ^
IndentationError: expected an indented block
>>>
>>> for i in grp_data_dev1:
...     print(i)
...
ns=2;s=simu.Device1.Group1.Meter
ns=2;s=simu.Device1.Group1.Press
ns=2;s=simu.Device1.Group1.temp
>>> for i in grp_data_dev1:
... ...     print(client.get_node(i))
  File "<stdin>", line 2
    ...     print(client.get_node(i))
    ^
IndentationError: expected an indented block
>>> for i in grp_data_dev1:
...     print(client.get_node(i))
...
ns=2;s=simu.Device1.Group1.Meter
ns=2;s=simu.Device1.Group1.Press
ns=2;s=simu.Device1.Group1.temp
>>> for i in grp_data_dev1:
...     nodes = client.get_node(i)
... for j in nodes:
  File "<stdin>", line 3
    for j in nodes:
    ^
SyntaxError: invalid syntax
>>> nodes
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
NameError: name 'nodes' is not defined
>>> for i in grp_data_dev1:
...     nodes = client.get_node(i)
...
>>> nodes
Node(StringNodeId(ns=2;s=simu.Device1.Group1.temp))
>>> for i in grp_data_dev1:
...     nodes = client.get_node(i)
...     print(nodes)
...
ns=2;s=simu.Device1.Group1.Meter
ns=2;s=simu.Device1.Group1.Press
ns=2;s=simu.Device1.Group1.temp
>>> nodes
Node(StringNodeId(ns=2;s=simu.Device1.Group1.temp))
>>> for i in grp_data_dev1:
...      nodes = client.get_node(i)
...      res = nodes.get_value())
  File "<stdin>", line 3
    res = nodes.get_value())
                           ^
TabError: inconsistent use of tabs and spaces in indentation
>>> for i in grp_data_dev1:
... ...      nodes = client.get_node(i)
  File "<stdin>", line 2
    ...      nodes = client.get_node(i)
    ^
IndentationError: expected an indented block
>>> ...      res = nodes.get_value()
  File "<stdin>", line 1
    ...      res = nodes.get_value()
             ^
SyntaxError: invalid syntax
>>> for i in grp_data_dev1:
... ...      nodes = client.get_node(i)
  File "<stdin>", line 2
    ...      nodes = client.get_node(i)
    ^
IndentationError: expected an indented block
>>> for i in grp_data_dev1:
...     nodes = client.get_node(i)
...     res = nodes.get_value()
...     print(res)
...
1152
0
1151
>>> for i in grp_data_dev1:
... ...     nodes = client.get_node(i)
  File "<stdin>", line 2
    ...     nodes = client.get_node(i)
    ^
IndentationError: expected an indented block
>>> ...     res = nodes.get_value()
  File "<stdin>", line 1
    ...     res = nodes.get_value()
            ^
SyntaxError: invalid syntax
>>> for i in grp_data_dev1:
...     nodes = client.get_node(i)
...     nodes.set_value(200)
...
Traceback (most recent call last):
  File "<stdin>", line 3, in <module>
  File "C:\Users\New\AppData\Local\Programs\Python\Python38\lib\site-packages\opcua\common\node.py", line 217, in set_value
    self.set_attribute(ua.AttributeIds.Value, datavalue)
  File "C:\Users\New\AppData\Local\Programs\Python\Python38\lib\site-packages\opcua\common\node.py", line 263, in set_attribute
    result[0].check()
  File "C:\Users\New\AppData\Local\Programs\Python\Python38\lib\site-packages\opcua\ua\uatypes.py", line 218, in check
    raise UaStatusCodeError(self.value)
opcua.ua.uaerrors._auto.BadWriteNotSupported: "The server does not support writing the combination of value, status and timestamps provided."(BadWriteNotSupported)
>>> node = client.get_node("ns=2;s=simu.Device1.Group1.Meter")
>>> node.get_value()
1948
>>> node.set_value(23)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\New\AppData\Local\Programs\Python\Python38\lib\site-packages\opcua\common\node.py", line 217, in set_value
    self.set_attribute(ua.AttributeIds.Value, datavalue)
  File "C:\Users\New\AppData\Local\Programs\Python\Python38\lib\site-packages\opcua\common\node.py", line 263, in set_attribute
    result[0].check()
  File "C:\Users\New\AppData\Local\Programs\Python\Python38\lib\site-packages\opcua\ua\uatypes.py", line 218, in check
    raise UaStatusCodeError(self.value)
opcua.ua.uaerrors._auto.BadWriteNotSupported: "The server does not support writing the combination of value, status and timestamps provided."(BadWriteNotSupported)
>>> node.set_writable()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\New\AppData\Local\Programs\Python\Python38\lib\site-packages\opcua\common\node.py", line 227, in set_writable
    self.set_attr_bit(ua.AttributeIds.AccessLevel, ua.AccessLevel.CurrentWrite)
  File "C:\Users\New\AppData\Local\Programs\Python\Python38\lib\site-packages\opcua\common\node.py", line 236, in set_attr_bit
    self.set_attribute(attr, val)
  File "C:\Users\New\AppData\Local\Programs\Python\Python38\lib\site-packages\opcua\common\node.py", line 263, in set_attribute
    result[0].check()
  File "C:\Users\New\AppData\Local\Programs\Python\Python38\lib\site-packages\opcua\ua\uatypes.py", line 218, in check
    raise UaStatusCodeError(self.value)
opcua.ua.uaerrors._auto.BadWriteNotSupported: "The server does not support writing the combination of value, status and timestamps provided."(BadWriteNotSupported)
>>> node.set_value(23)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\New\AppData\Local\Programs\Python\Python38\lib\site-packages\opcua\common\node.py", line 217, in set_value
    self.set_attribute(ua.AttributeIds.Value, datavalue)
  File "C:\Users\New\AppData\Local\Programs\Python\Python38\lib\site-packages\opcua\common\node.py", line 263, in set_attribute
    result[0].check()
  File "C:\Users\New\AppData\Local\Programs\Python\Python38\lib\site-packages\opcua\ua\uatypes.py", line 218, in check
    raise UaStatusCodeError(self.value)
opcua.ua.uaerrors._auto.BadWriteNotSupported: "The server does not support writing the combination of value, status and timestamps provided."(BadWriteNotSupported)
>>> node = client.get_node("ns=2;s=simu.Device1.tG")
>>> node.get_value()
49775
>>> node.get_value()
49811
>>> node.set_writable()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\New\AppData\Local\Programs\Python\Python38\lib\site-packages\opcua\common\node.py", line 227, in set_writable
    self.set_attr_bit(ua.AttributeIds.AccessLevel, ua.AccessLevel.CurrentWrite)
  File "C:\Users\New\AppData\Local\Programs\Python\Python38\lib\site-packages\opcua\common\node.py", line 236, in set_attr_bit
    self.set_attribute(attr, val)
  File "C:\Users\New\AppData\Local\Programs\Python\Python38\lib\site-packages\opcua\common\node.py", line 263, in set_attribute
    result[0].check()
  File "C:\Users\New\AppData\Local\Programs\Python\Python38\lib\site-packages\opcua\ua\uatypes.py", line 218, in check
    raise UaStatusCodeError(self.value)
opcua.ua.uaerrors._auto.BadWriteNotSupported: "The server does not support writing the combination of value, status and timestamps provided."(BadWriteNotSupported)
>>> node.set_value(21)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\New\AppData\Local\Programs\Python\Python38\lib\site-packages\opcua\common\node.py", line 217, in set_value
    self.set_attribute(ua.AttributeIds.Value, datavalue)
  File "C:\Users\New\AppData\Local\Programs\Python\Python38\lib\site-packages\opcua\common\node.py", line 263, in set_attribute
    result[0].check()
  File "C:\Users\New\AppData\Local\Programs\Python\Python38\lib\site-packages\opcua\ua\uatypes.py", line 218, in check
    raise UaStatusCodeError(self.value)
opcua.ua.uaerrors._auto.BadWriteNotSupported: "The server does not support writing the combination of value, status and timestamps provided."(BadWriteNotSupported)
>>> node
Node(StringNodeId(ns=2;s=simu.Device1.tG))
>>> node.get_value()
301.25
>>> node.set_value(34.99)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\New\AppData\Local\Programs\Python\Python38\lib\site-packages\opcua\common\node.py", line 217, in set_value
    self.set_attribute(ua.AttributeIds.Value, datavalue)
  File "C:\Users\New\AppData\Local\Programs\Python\Python38\lib\site-packages\opcua\common\node.py", line 263, in set_attribute
    result[0].check()
  File "C:\Users\New\AppData\Local\Programs\Python\Python38\lib\site-packages\opcua\ua\uatypes.py", line 218, in check
    raise UaStatusCodeError(self.value)
opcua.ua.uaerrors._auto.BadWriteNotSupported: "The server does not support writing the combination of value, status and timestamps provided."(BadWriteNotSupported)
>>> dv = ua.DataValue(ua.Variant(1.2, ua.VariantType.Float))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
NameError: name 'ua' is not defined
>>> from opcua import ua
>>> dv = ua.DataValue(ua.Variant(1.2, ua.VariantType.Float))
>>> dv.ServerTimestamp = None
>>> dv.SourceTimestamp = None
>>> node.set_value(34.6)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\New\AppData\Local\Programs\Python\Python38\lib\site-packages\opcua\common\node.py", line 217, in set_value
    self.set_attribute(ua.AttributeIds.Value, datavalue)
  File "C:\Users\New\AppData\Local\Programs\Python\Python38\lib\site-packages\opcua\common\node.py", line 263, in set_attribute
    result[0].check()
  File "C:\Users\New\AppData\Local\Programs\Python\Python38\lib\site-packages\opcua\ua\uatypes.py", line 218, in check
    raise UaStatusCodeError(self.value)
opcua.ua.uaerrors._auto.BadWriteNotSupported: "The server does not support writing the combination of value, status and timestamps provided."(BadWriteNotSupported)
>>> node.set_value(dv)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\New\AppData\Local\Programs\Python\Python38\lib\site-packages\opcua\common\node.py", line 217, in set_value
    self.set_attribute(ua.AttributeIds.Value, datavalue)
  File "C:\Users\New\AppData\Local\Programs\Python\Python38\lib\site-packages\opcua\common\node.py", line 263, in set_attribute
    result[0].check()
  File "C:\Users\New\AppData\Local\Programs\Python\Python38\lib\site-packages\opcua\ua\uatypes.py", line 218, in check
    raise UaStatusCodeError(self.value)
opcua.ua.uaerrors._auto.BadTypeMismatch: "The value supplied for the attribute is not of the same type as the attribute"s value."(BadTypeMismatch)
>>> dv = ua.DataValue(ua.Variant(1.2, ua.VariantType.Long))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\New\AppData\Local\Programs\Python\Python38\lib\enum.py", line 341, in __getattr__
    raise AttributeError(name) from None
AttributeError: Long
>>>     print(client.get_node(i))